pkg/bpmn/model/bpmn20/core.go

```diff
@@ -33,27 +39,81 @@ type TBaseElement struct {
 	Documentation []TDocumentation `xml:"documentation"`
 }
 
-func (baseElement TBaseElement) GetId() string {
+func (baseElement *TBaseElement) GetId() string {
 	return baseElement.Id
 }
-func (baseElement TBaseElement) GetDocumentation() []Documentation {
+func (baseElement *TBaseElement) GetDocumentation() []Documentation {
 	var docs []Documentation
 	for _, doc := range baseElement.Documentation {
-		docs = append(docs, doc)
+		docs = append(docs, &doc)
 	}
 	return docs
 }
 
+func collectBaseElements(element interface{}, refs *map[string]BaseElement) error {
```

Comment:
I would like to see some tests for this where you build out more complicated struct in test itself and try to parse it.

Reply:
It doesn't have to be a complex process...I meant that I would like to have a test that catches some corner cases ex: variable is/is not a pointer, variable is an interface (idk if we want to handle this but at least have a panic for that), array of pointer/direct vars etc...